pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest.java

```diff
@@ -3694,4 +3695,22 @@ public void testPermissionsAllowAclChangesOnNonExistentTopics() {
             pulsar.getConfiguration().setAllowAclChangesOnNonExistentTopics(false);
         }
     }
+
+    @Test
+    public void testRecreatePartitionedTopicAfterMetadataLoss()
+            throws PulsarAdminException, ExecutionException, InterruptedException {
+        String namespace = "prop-xyz/ns1/";
+        final String random = UUID.randomUUID().toString();
+        final String topic = "persistent://" + namespace + random;
+        admin.topics().createPartitionedTopic(topic, 5);
+
+        // Delete the topic metadata.
+        pulsar.getPulsarResources().getNamespaceResources().getPartitionedTopicResources()
+                .deletePartitionedTopicAsync(TopicName.get(topic)).get();
+        List<String> partitionedTopicList = admin.topics().getPartitionedTopicList(namespace);
+        assertThat(partitionedTopicList).doesNotContain(topic);
+
+        // Create the partitioned topic again.
+        admin.topics().createPartitionedTopic(topic, 5);
```

Comment:
Why check the metadata completeness again?

Reply:
He removed the partitioned metadata at line `3708` to mock an issue that someone did the wrong operation on the metadata store.

After that, he can reproduce the scenario he described in the Motivation section.

Reply:
Reproduce the metadata loss:

1. Create the partitioned topic.
2. Delete the partitioned topic metadata.
3. Recreate the partitioned topic.